lib/_stream_readable.js

```diff
@@ -311,7 +311,8 @@ Readable.prototype.setEncoding = function(enc) {
   if (!StringDecoder)
     StringDecoder = require('string_decoder').StringDecoder;
   this._readableState.decoder = new StringDecoder(enc);
-  this._readableState.encoding = enc;
+  // if setEncoding(null), decoder.encoding = 'utf8'
+  this._readableState.encoding = this._readableState.decoder.encoding;
```

Comment:
```javascript
const readStream = new ReadableStream();
readStream.setEncoding(null);
// readStream._readableState.decoder.encoding equals 'utf8'
// readStream._readableState.encoding equals null;
```
`_readableState.encoding` should be same as  `this._readableState.decoder.encoding`